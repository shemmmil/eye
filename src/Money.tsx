import React, { useState } from 'react';
import styled from 'styled-components';
import { Modal, Button } from 'antd';
import { PlusOutlined } from '@ant-design/icons';

const Container = styled.span`
  display: inline-flex;
  margin-left: 6px;
  margin-right: 32px;
`;

const Title = styled.h5`
  font-size: 15px;
  font-weight: 700;
  display: block;
  margin-top: 0;
  margin-bottom: 6px;
`;

const Subtitle = styled.span`
  color: #888;
  font-size: 13px;
`;

export const Money: React.FC = () => {
  const [visible, setVisible] = useState<boolean>(false);

  const showModal = (): void => setVisible(true);

  const handleCancel = (): void => setVisible(false);

  return (
    <>
      <Button shape="circle" icon={<PlusOutlined />} onClick={showModal} />
      <Container>0&nbsp;₽</Container>
      <Modal title="Остаток на счете: 0 ₽" visible={visible} onCancel={handleCancel}>
        <Title>Пополнение счета</Title>
        <Subtitle>Выберите один из рекомендованных вариантов пополнения</Subtitle>
      </Modal>
    </>
  );
};
