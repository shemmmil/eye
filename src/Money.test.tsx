import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import { Money } from './Money';

describe('Money Render', () => {
  it('Money render correctly', () => {
    const wrapper = mount(<Money />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
