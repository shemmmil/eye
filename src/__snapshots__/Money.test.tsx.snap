// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Money Render Money render correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: 6px;
  margin-right: 32px;
}

<Component>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    icon={<ForwardRef(PlusOutlined) />}
    loading={false}
    onClick={[Function]}
    shape="circle"
  >
    <Wave>
      <button
        className="ant-btn ant-btn-circle ant-btn-icon-only"
        onClick={[Function]}
        type="button"
      >
        <ForwardRef(PlusOutlined)>
          <AntdIcon
            icon={
              Object {
                "icon": Object {
                  "attrs": Object {
                    "focusable": "false",
                    "viewBox": "64 64 896 896",
                  },
                  "children": Array [
                    Object {
                      "attrs": Object {},
                      "children": Array [
                        Object {
                          "attrs": Object {},
                          "tag": "style",
                        },
                      ],
                      "tag": "defs",
                    },
                    Object {
                      "attrs": Object {
                        "d": "M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z",
                      },
                      "tag": "path",
                    },
                    Object {
                      "attrs": Object {
                        "d": "M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z",
                      },
                      "tag": "path",
                    },
                  ],
                  "tag": "svg",
                },
                "name": "plus",
                "theme": "outlined",
              }
            }
          >
            <span
              aria-label="plus"
              className="anticon anticon-plus"
              role="img"
            >
              <IconReact
                className=""
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {},
                          "children": Array [
                            Object {
                              "attrs": Object {},
                              "tag": "style",
                            },
                          ],
                          "tag": "defs",
                        },
                        Object {
                          "attrs": Object {
                            "d": "M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z",
                          },
                          "tag": "path",
                        },
                        Object {
                          "attrs": Object {
                            "d": "M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "plus",
                    "theme": "outlined",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="plus"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  key="svg-plus"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <defs
                    key="svg-plus-svg-0"
                  >
                    <style
                      key="svg-plus-svg-0-defs-0"
                    />
                  </defs>
                  <path
                    d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                    key="svg-plus-svg-1"
                  />
                  <path
                    d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
                    key="svg-plus-svg-2"
                  />
                </svg>
              </IconReact>
            </span>
          </AntdIcon>
        </ForwardRef(PlusOutlined)>
      </button>
    </Wave>
  </Button>
  <styled.span>
    <span
      className="c0"
    >
      0 ₽
    </span>
  </styled.span>
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    title="Остаток на счете: 0 ₽"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <Component
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <ForwardRef(CloseOutlined)
            className="ant-modal-close-icon"
          />
        </span>
      }
      confirmLoading={false}
      footer={
        <LocaleReceiver
          componentName="Modal"
          defaultLocale={
            Object {
              "cancelText": "Cancel",
              "justOkText": "OK",
              "okText": "OK",
            }
          }
        >
          [Function]
        </LocaleReceiver>
      }
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      prefixCls="ant-modal"
      title="Остаток на счете: 0 ₽"
      transitionName="zoom"
      visible={false}
      width={520}
      wrapClassName=""
    >
      <PortalWrapper
        visible={false}
      />
    </Component>
  </Modal>
</Component>
`;
